posts/comparing-java-lts-releases.md

```diff
@@ -0,0 +1,128 @@
+```
+Modules belong to a wider initiative known as Project Jigsaw, which again extends beyond the scope of this blog post.  For information on modules, how they work, and Project Jigsaw, [check out this article](http://openjdk.java.net/projects/jigsaw/quick-start)
+
+# Java 10 Non-LTS
+Java 9 was the release that re-defined how Java programs could be organized and developed.  Java 10 was the release that built upon the foundation laid in Java 9.  The most subtle, but most important feature of this release were the performance enhancements introduced to the jshell tool.  The jshell tool was a powerful addition in Java 9, but it took awhile for it to get going.  Java 10 kicked it into gear a lot faster than Java 9 did, introducing performance enhancements that made the tool more desirable to use for developers.  
+
+Additionally, Java 10 improved upon Java 9's new Garbage Collector, the Garbage-First Garbage Collector.  Java 9's G1GC changed the way Java performed garbage collecting by partitioning its heap into multiple small heaps, whereas traditionally the heap would be split into three chunks.  G1GC also introduced multi-threaded processing on the garbage collector for all events save a "stop-the world," a point in time during garbage collection where program execution halts and gives the garbage collector time to assess the state of the application.  "Stop-the-world" events are unaviodable in garbage collection.  Java 10 took the multi-threaded approach perfected by Java 9 and applied it to the "stop-the-world" event, thus making the G1GC fully concurrent.  Even though the G1GC is fully concurrent and therefore faster than prior garbage collectors on average, it is worth noting it takes a larger memory footprint because of the way it chops the heap into more chunks.
+
+The big language feature introduced in Java 10 is the `var` keyword.  The `var` keyword intoduces implicitly typed variables to Java.  The concept of the implicit type has been around forever; Python and Perl use implicit typing out of the box, as does C# (with reflection or the `var` keyword).  Java has always traditionally been a statically typed language; the closest to dynamic typing with Java you could get before version 10 was with Generic classes and runtime casting.  With the introduction of the `var` keyword though, we can infer an object's type at compile time, making code easier to read.  Java is still a statically typed language, nothing will change that.  But the introduciton of the `var` keyword extends the usability of Java to use cases where developers may choose a dynamically typed language by default.  This is an exciting potential use-case for Java.  The use of the `var` keyword, combined with tools like the `jshell`, propel Java into the functional programming space.  That being said, I would caution excited fans of inferred types to use `var` only when necessary.  The following is a perfect example:
```

Comment:
"the closest to dynamic typing with Java" -> I would rephrase that because Java didn't get closer to dynamic typing with the "var" keyword. Dynamic typing is when typing is done at runtime. Type inference is done at compile time, there is no type resolution at runtime.